CleverPush/Source/CleverPushInstance.m

```diff
@@ -1514,24 +1515,21 @@ - (void)handleNotificationOpened:(NSDictionary*)payload isActive:(BOOL)isActive
             [CPAppBannerModuleInstance setCurrentVoucherCodePlaceholder:voucherCodesByAppBanner];
         }
 
-        [self showAppBanner:[notification valueForKey:@"appBanner"] channelId:[payload cleverPushStringForKeyPath:@"channel._id"] notificationId:notificationId];
-    }
-
-    CPNotificationOpenedResult * result = [[CPNotificationOpenedResult alloc] initWithPayload:payload action:action];
-
-    if (!channelId) { // not init
-        pendingOpenedResult = result;
+        [self showAppBanner:[notification valueForKey:@"appBanner"] channelId:[payloadMutable cleverPushStringForKeyPath:@"channel._id"] notificationId:notificationId];
     }
 
     if (notification != nil && [notification objectForKey:@"url"] != nil && ![[notification objectForKey:@"url"] isKindOfClass:[NSNull class]] && [[notification objectForKey:@"url"] length] != 0) {
-        NSURL *url = [NSURL URLWithString:[notification objectForKey:@"url"]];
         if ([notification objectForKey:@"autoHandleDeepLink"] != nil && ![[notification objectForKey:@"autoHandleDeepLink"] isKindOfClass:[NSNull class]] && [[notification objectForKey:@"autoHandleDeepLink"] boolValue]) {
-            if ([[UIApplication sharedApplication] canOpenURL:url]) {
-                [[UIApplication sharedApplication] openURL:url options:@{} completionHandler:nil];
-            } 
+            payloadMutable = [self handleActionInNotification:notification withAction:action payloadMutable:payloadMutable];
```

Comment:
Now this is only called when `autoHandleDeepLink` is true, which is probably not what we want. We want to override notification.url with the action.url also in the NotificationOpenedResult if autoHandleDeepLink is not used at all.